### ISSUE TITLE
Attention in the discriminator vs generator

### ISSUE THREAD

**Issue by dorarad** [GiorgiaAuroraAdorni/gansformer-reproducibility-challenge#4]
Dear Giorgia, Felix and Stefano, 
Thank you very much for the interest in the GANformer model and in working on reproducing it! 

I just read the reproducibility report and wanted to comment regarding the attention in the generator vs. discriminator. 
In earlier stages of the model development, I have explored the use of bipartite attention in both the generator and the discriminator, and similarly to the observations you've had, as I kept working on it, I noticed that indeed the model performs better when incorporating attention to the generator only, and so the pre-trained models + default command-line settings of the public repository reflect that. 

In the GANformer2 paper we released later last year, we also followed this similar design (of using bipartite attention over the generator only). The paragraph about the discriminator using bipartite attention too at the bottom right of GANformer's paper page 7 should have been removed and stayed there due to a mistake on my side. I updated the paper today to address that and it should become public through arxiv later today! 

Also wanted to mention that in the report where it compared epsilon values, note that "epsilon" (in the code) doesn't stand for learning rate, but rather it's used as a small value that is added by the optimizer for stability. It was set to 1e-8 in both StyleGAN2 and GANformer2 repositories. Meanwhile, the learning rate in the GANformer's repository is 0.001 in line with the paper. (For the other optimizer settings of beta1, beta2 where the Supp. says "beta1=0.9, beta1=0.999", I have fixed that to "beta1=0.0, beta2=0.9" to comply with the repository).

Thanks very much and best wishes,
- Drew
